datasource/server.go

```diff
@@ -4,11 +4,19 @@ import (
 	"net/http"
 )
 
-func LinkHandler(w http.ResponseWriter, r *http.Request, data []byte) {
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
-	_, err := w.Write(data)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+func ColdestTemperatureLocationHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Let this function accept a third parameter that is a function. When you call the function, you get data.

In production code, this function is "getData" - (which should be renamed also)

In test code, we can create a new function that returns test data

Reply:
Ok great, I think I know what and how to do this! Just to double check: I will use getData as input in the production code and getMockedData (already exists) in test code. However I would need to change getData so that it does no longer take useMocked bool as input. Then the two functions would have the same signature and I could use them as input to ColdestTemperatureLocationHandler. Would that be a good solution?